### ISSUE TITLE
Account redesign

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/py-evm#191]
1. We need to support the account storage redesign.  This should be a configurable `StateDB` class that implements the following structure for accessing account information.
2. [A two-layer account trie inside a single-layer trie](https://ethresear.ch/t/a-two-layer-account-trie-inside-a-single-layer-trie/210)
3. [doc.md](https://github.com/ethereum/sharding/blob/develop/docs/doc.md#two-layer-trie-redesign)

```python
class State():
    def __init__(self, state_root, env, db):
        self.state_root, self.env. self.db = state_root, env, db
        cache = {}
        journal = []

    def get_balance(self, acct):
        key = sha3(acct) + b'\x00'
        if key not in self.cache:
            self.cache[key] = _trie_get(self.db, self.state_root)
        return self.cache[key]

    def get_nonce(self, acct): key = sha3(acct) + b'\x01' ...

    def get_code(self, acct): key = sha3(acct) + b'\x02' ...

    def get_storage(self, acct, k): key = sha3(acct) + b'\x03' + k ....
```

**Comment by hwwhww** [comment]
@vbuterin Should we update the length of address in this ticket?
I remember is `20 bytes -> 24 bytes`?

**Comment by hwwhww** [comment]
This ticket is assigned to @jannikluhn

**Comment by jannikluhn** [comment]
Closed with #213
